tools/commit_slider/case_1.md

```diff
@@ -0,0 +1,316 @@
+    simpleSubstitute(cfg, "actualAppCmd", "$.runConfig.secondAppCmd", "$.appCmd")
+
+    # run second app
+    checkOut = fetchAppOutput(cfg, commit)
+    commitLogger.info(checkOut)
+    foundThroughput = re.search(
+        self.outPattern, checkOut, flags=re.MULTILINE
+    ).group(1)
+    self.secondThroughput = foundThroughput
+    self.secondModel = cfg['appCmd']
+
+    # add commit data to commit path
+    pc = Mode.CommitPath.PathCommit(
+        commit,
+        Mode.CommitPath.CommitState.DEFAULT
+    )
+    self.setOutputInfo(pc)
+    self.commitPath.accept(self.traversal, pc)
+
+def  printResult(self):
+    # formatting of full output
+    for  pathcommit  in  self.commitPath.getList():
+        commitInfo = self.getCommitInfo(pathcommit)
+        print(commitInfo)
+        self.outLogger.info(commitInfo)
+
+def  setOutputInfo(self, pathCommit):
+    # populating of commit path
+    pathCommit.firstThroughput = self.firstThroughput
+    pathCommit.firstModel = self.firstModel
+    pathCommit.secondThroughput = self.secondThroughput
+    pathCommit.secondModel = self.secondModel
+
+def  getCommitInfo(self, commit):
+    return  "{hash}, throughput_1 = {t1}, throughput_2 = {t2}".format(
+        hash=commit.cHash,
+        t1=commit.firstThroughput,
+        t2=commit.secondThroughput)
+```
+### Configuration
+```
+supposedly we pass both commandlines launching models in two parameters cfg["runConfig"]["firstAppCmd"] and cfg["runConfig"]["secondAppCmd"], i.e.
+cfg = { 
+    "appCmd" : "{actualAppCmd}",
+    "appPath": "<path_to_benchmark_app>",
+    "gitPath" : "<path_to_git_repository>",
+    "buildPath" : "<path_to_build_dir>",
+        "runConfig" : {
+            "commitList" : {
+                "explicitList" : [ "{start}", "{end}" ]
+            },
+            "mode" : "crossCheck",
+            "firstAppCmd" : "./benchmark_app -m model_1.xml --other_params",
+            "secondAppCmd" : "./benchmark_app -m model_2.xml --other_params",
+            "traversal" : "bruteForce"
+        }
+}
+```
+### Output
+```
+python3 commit_slider.py -cfg cfg.json
+
+<hash_1>, throughput_1 = 500.0, throughput_2 = 500.0
+<hash_2>, throughput_1 = 1000.0, throughput_2 = 1000.0
+
+```
+We've got rid of full output and learned to provide only necessary information, but we still limited by verbose configuration-style API and still provide raw output information.
+
+## Third Version: Specific Template
+###  API design
+We want to provide for user 2 ways to set up command lines for comparation. ```'appCmd' : ['./benchmark_app --parameters_1', './benchmark_app --parameters_2']``` for detailed parameters and simplified ```'par_1' : 'parameter_1', 'par_2' : 'parameter_2'``` supposing that substitute the placeholder ```'appCmd' : 'prefix {actualPar} postfix'```. The example of this approach may be ```'par_1' : 'model_1.xml', 'par_2' : 'model_2.xml' : 'parameter_2', 'appCmd' : './benchmark_app -m {actualPar} -i input.png'```
+  
+### Custom template
```

Comment:
Can we define what mode and template means? Or refer to these terms from the README

Reply:
I didn't find any information about template in README. If it is missed, can we add a few words on that, and refer to them here?

Reply:
Template description added to common_architecture + reference